src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -82,12 +85,12 @@ public Http2Stream(HttpRequestMessage request, Http2Connection connection, int s
                 _connection = connection;
                 _streamId = streamId;
 
-                _state = StreamState.ExpectingStatus;
+                _requestCompletionState = StreamCompletionState.InProgress;
+                _responseCompletionState = StreamCompletionState.InProgress;
 
-                _request = request;
-                _shouldSendRequestBody = true;
+                _responseProtocolState = ResponseProtocolState.ExpectingStatus;
 
-                _disposed = false;
+                _request = request;
```

Comment:
Nit: move this to be the very first line, so that the first three lines map to assigning the first three arguments to fields?

Reply:
Will do